.github/actions/setup-go-tip/action.yml

```diff
@@ -8,16 +8,32 @@ runs:
       shell: bash
       run: |
         set -euo pipefail
-        tip=$(git ls-remote https://github.com/golang/go.git HEAD | awk '{print $1;}')
-        echo "Go Tip version: ${tip}"
-        curl -fsSL https://storage.googleapis.com/go-build-snap/go/linux-amd64/${tip}.tar.gz -o gotip.tar.gz
-        echo "Downloaded bundle:"
-        ls -lah gotip.tar.gz
+        retries=10
+        wait_time=30
+        success=false
+        for ((i=1; i<=retries; i++)); do
+          tip=$(git ls-remote https://github.com/golang/go.git HEAD | awk '{print $1;}')
+          echo "Go Tip version: ${tip}"
+          if curl_output=$(curl -fsSL -w "%{http_code}" -o gotip.tar.gz https://storage.googleapis.com/go-build-snap/go/linux-amd64/${tip}.tar.gz); then
+            echo "Downloaded bundle:"
+            ls -lah gotip.tar.gz
+            success=true
+            break
+          else
```

Comment:
don't need else nesting, since you break out in if()

Reply:
done!